src/lit-element.ts

```diff
@@ -161,18 +162,28 @@ export class LitElement extends PropertiesMixin
    * @param {*} _props Current element properties
    * @returns {TemplateResult} Must return a lit-html TemplateResult.
    */
-  protected render(_props: object): TemplateResult {
+  protected _render(_props: object): TemplateResult {
     throw new Error('render() not implemented');
   }
 
+  protected _shouldRender(_props: object, _changedProps: object, _prevProps: object) {
+    return true
+  }
+
+  protected _willRender(_props: object, _changedProps: object, _prevProps: object) {}
+
+  protected _applyRender(result: TemplateResult, _root: Node) {
```

Comment:
Need API docs for `_shouldRender`, `_willRender`, `_applyRender`.

Think we also need a section in README on lifecycle, and what one might do in each callback (i.e. what's its intended use case/purpose).  The one I'm most unclear on is `_willRender`.  What good thing can you do here that you can't do in `_render`?